crypto/s2n_hash.c

```diff
@@ -95,7 +95,7 @@ int s2n_hash_is_available(s2n_hash_algorithm alg)
         is_available = 1;
         break;
     default:
-        S2N_ERROR(S2N_ERR_HASH_INVALID_ALGORITHM);
+        is_available = 0;
```

Comment:
Instead of tracking a local variable, why not just return the result?

Reply:
then we will have multiple return statements instead of one.

Reply:
What's wrong with that? There are plenty of s2n functions that have many return statements, and doing so often makes the code clearer.

Reply:
Well, there is nothing wrong in doing that.
I was just trying to follow the general trend in this file for ex: s2n_hash_hmac_alg, s2n_hash_digest_size, s2n_hash_block_size. 
And about readability, it doesn't bother me much in this case. I am fine with having multiple return statements.

Reply:
I actually think its slightly clearer to have multiple returns for a function of this type - you can see in a single place what its returning and why, rather than having to see what the initial value of the variable was, and whether it got overridden. But I acknowledge that this is a aesthetic judgement.

Reply:
removed local var and added return.